tests/integration/test_s3_table_function_with_http_proxy_list/test.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import time
+
+import pytest
+from helpers.cluster import ClickHouseCluster
+
+
+@pytest.fixture(scope="module")
+def cluster():
+    try:
+        cluster = ClickHouseCluster(__file__)
+
+        cluster.add_instance(
+            "proxy_list_node",
+            main_configs=["configs/config.d/proxy_list.xml"],
+            with_minio=True,
+        )
+
+        logging.info("Starting cluster...")
+        cluster.start()
+        logging.info("Cluster started")
+
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+def check_proxy_logs(cluster, proxy_instance, http_methods={"POST", "PUT", "GET"}):
+    for i in range(10):
+        logs = cluster.get_container_logs(proxy_instance)
+        # Check with retry that all possible interactions with Minio are present
+        for http_method in http_methods:
+            if logs.find(http_method + " http://minio1") >= 0:
+                return
+            time.sleep(1)
```

Comment:
Why do you need a `sleep` here?

Reply:
I honestly do not know, this was already there: https://github.com/ClickHouse/ClickHouse/blob/master/tests/integration/test_s3_with_proxy/test.py#L47

Reply:
Introduced by a "fix flaky test" [PR](https://github.com/ClickHouse/ClickHouse/pull/15307) by: @alesapin